index.js

```diff
@@ -11,27 +11,29 @@ const defaultApiVersion = 'v1'
 
 class Client {
   constructor (auth, inputApiUrl = defaultApiUrl) {
-    if (auth === undefined) {
+    if (typeof auth !== 'object' || Array.isArray(auth)) {
```

Comment:
I don't get it. The equivalent of the old check is just `if (!auth)`. If we want to check that `auth` is a plain object, I would add `lodash` library as dependency and use https://lodash.com/docs/4.17.11#isPlainObject check (that will be more readable, and no need to worry that we miss some edge-cases here).

Reply:
I agree, `if (!auth)` looks ok to me here.

Reply:
Ok, then we'll just check that `auth` exists and assume that `auth` is an object.